poc_iot_verifier/src/poc_report.rs

```diff
@@ -86,21 +98,49 @@ impl Report {
         .map_err(Error::from)
     }
 
+    pub async fn delete_report<'c, 'q, E>(executor: E, id: &'q Vec<u8>) -> Result
+    where
+        E: sqlx::Executor<'c, Database = sqlx::Postgres> + Clone,
+    {
+        sqlx::query(
+            r#"
+            delete from poc_report
+            where id = $1
+            "#,
+        )
+        .bind(id)
+        .execute(executor.clone())
+        .await
+        .map(|_| ())
+        .map_err(Error::from)
+    }
+
     pub async fn get_next_beacons<'c, E>(executor: E) -> Result<Vec<Self>>
     where
         E: sqlx::Executor<'c, Database = sqlx::Postgres>,
     {
         sqlx::query_as::<_, Self>(
             r#"
-            select * from poc_report
-            where report_type = 'beacon' and status = 'pending'
-            and report_timestamp < (NOW() - INTERVAL '$1 MINUTES')
+            select  poc_report.id,
+                    poc_report.remote_entropy,
+                    poc_report.packet_data,
+                    poc_report.report_data,
+                    poc_report.report_type,
+                    poc_report.status,
+                    poc_report.attempts,
+                    poc_report.report_timestamp,
+                    poc_report.last_processed,
+                    poc_report.created_at
+            from poc_report
+            left join entropy on poc_report.remote_entropy=entropy.data
+            where poc_report.report_type = 'beacon' and status = 'pending'
+            and entropy.timestamp < (NOW() - INTERVAL '$1 SECONDS')
```

Comment:
this query will retrieve beacons which have an entropy timestamp  which are **older** than 90 seconds or that defined by ENTROPY_LIFESPAN.  As such it is asking for reports for which the entropy has expired and thus ready for verification.   We arent interested in processing reports until after ENTROPY_LIFESPAN has expired.

As to retries and i assume you mean client retries, all these values will need considered together....90 seconds is what i have atm for the entropy lifespan but that is not to say it will remain as that

Reply:
Gotcha; for some reason i read the `<` as being within a range. 👍